src/app_service/service/community/service.nim

```diff
@@ -1203,6 +1240,24 @@ QtObject:
 
     self.joinedCommunities[communityId] = community
 
+  proc cancelRequestToJoinCommunity*(self: Service, communityId: string) =
+    try:
+      var i = 0
+      for communityRequest in self.myCommunityRequests:
+        if (communityRequest.communityId == communityId):
+          let response = status_go.cancelRequestToJoinCommunity(communityRequest.id)
+          if (not response.error.isNil):
+            let msg = response.error.message & " communityId=" & communityId
+            error "error while cancel membership request ", msg
+            return
+          self.myCommunityRequests.delete(i)
+
+        i.inc()
+      
+      self.events.emit(SIGNAL_REQUEST_TO_JOIN_COMMUNITY_CANCELED, Args())
```

Comment:
@borismelnik this signal is for the future ? It's not handled anywhere.

Reply:
yes